### PR TITLE
feat(editor): add sub-day time filters for collection date rules

### LINKED ISSUE #15368: [Feature Request]: enable collection filter < 3days
### Description

Story:
My users are a fast paced organization.  They need to quickly identify documents added or updated in timeframes tighter than 3 days.  

Proposal:
Create a filter `newer than` with `hours` and `minutes` fields which filters to select all objects newer than X hours and Y minutes.

Success:
Users are able to create a collection containing all documents created or updated in the past X hours and Y minutes.

### Use case

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### Is your content generated by AI?

- [ ] I confirm that the content I submitted was **not** generated by AI / **merely contained minimal** AI edits.

### PR DESCRIPTION
**Pull request by devansh9837**
Add hour and minute granularity to the collection filter system so users can filter docs by created/updated within the last N hours or minutes, not just days.

New filter methods: last-30-minutes, last-1-hour, last-3-hours, last-6-hours, last-12-hours, last-24-hours.

Bug fixes applied to the existing date filter:
- Move now into map() so long-lived subscriptions always use the real current time, not the subscription-creation time (stale now bug)
- Split isAfter into isAfterOrOn (inclusive, for relative ranges like last-N-days/this-week) and isStrictlyAfter/isStrictlyBefore (exclusive, for explicit after/before/between UI methods)
- Return empty set for incomplete between range instead of silently matching nothing
- Remove last-60-minutes and last-180-minutes aliases (identical to last-1-hour and last-3-hours, caused duplicate UI dropdown options)
- Add handleTimestampFilter for full epoch-ms precision on sub-day ranges (existing date-tuple approach truncates to day granularity)

Also adds 26 unit tests in date.spec.ts covering all filter methods, boundary conditions, the stale-now regression, and value encoding (numeric epoch, digit string, ISO string, unparseable garbage).

Closes #15368

## Summary by CodeRabbit

* **New Features**
  * Added date filters for the last 30 minutes, 1, 3, 6, 12, and 24 hours.
  * Added English labels for the new filter options.

* **Bug Fixes**
  * Improved accuracy for relative, explicit, and timestamp-based date ranges.
  * Relative ranges now recalculate as data changes.
  * Explicit date ranges require complete values and apply clearer boundary rules.
  * Removed redundant minute-based filter aliases.

### CODE REVIEW

**Comment by devansh9837** [comment]
Hi @darkskygit @pengx17 - could you take a look when you get a chance? This implements the feature requested in #15368 (sub-day time filters for collections) and also fixes 5 pre-existing bugs in the date filter logic. CodeRabbit pre-merge checks are running. Thanks!

File: packages/frontend/core/src/modules/collection-rules/impls/filters/date.ts [hunk 1]
```diff
@@ -62,30 +69,58 @@ export function basicDateFilter(
...
+            isStrictlyAfter(parsed, filterValues[0])
           );
         }
 
         if (method === 'before' && filterValues.length >= 1) {
           return handleDateRangeFilter(o, parsed =>
-            isBefore(parsed, filterValues[0])
+            isStrictlyBefore(parsed, filterValues[0])
           );
         }
```
[the remark this answers is not in the document]
Comment (devansh9837) [reply packages/frontend/core/src/modules/collection-rules/impls/filters/date.ts:104]: Fixed in commit 3cbf7de - after and before now return an empty Set when filterValues is empty, matching the between pattern.

**Comment by devansh9837** [comment]
Hi @EYHN @CatsJuice @pengx17 - this PR adds sub-day time filters (last hour, last 6 hours, today, yesterday) to the collection date rules in modules/collection-rules. Would appreciate a review when you have a moment. Thanks!

**Comment by devansh9837** [comment]
Hi @EYHN @CatsJuice @pengx17 @darkskygit - this PR adds sub-day time filters (last hour, last 6 hours, today, yesterday) to the collection date rules in modules/collection-rules. Would appreciate a review when you have a moment. Thanks!

**Comment by donqu1xotevincent** [comment]
Thanks for the contribution — the feature direction is right, the core implementation is reasonable, and the test coverage (boundary values, fake-timer regression test, value-encoding cases) is genuinely solid. I verified that `created-at.ts` and `updated-at.ts` both route through `basicDateFilter`, so the new methods do apply to Created/Updated, and the colocated `date.spec.ts` is picked up by the root vitest include pattern.

That said, a few things should be addressed before this is mergeable.

### 🔴 1. The PR describes removing aliases that never existed

The description says `last-60-minutes` and `last-180-minutes` were "removed" because they "caused duplicate UI dropdown options". I checked `canary`: **these keys have never existed in this repo** — not in `types.ts`, `date.ts`, or `DateFilterMethod`. They appear to be leftovers from an earlier iteration of this branch, not upstream history.

Please clean up:
- The three code comments referencing the "removal" (in `workspace-property-types/index.ts`, `date.ts`, and `types.ts`) — they'll send future readers hunting through git history for something that was never there.
- The two `removed alias methods throw` tests in `date.spec.ts` — they duplicate the existing `unknown method` test.
- The corresponding "bug fix" bullet in the PR description.

### 🔴 2. Silent behavior change for existing saved filters

On `canary`, `after` is `>=`, `before` is `<=`, and `between` is inclusive on both ends. This PR flips all three to strictly exclusive. That means:

- Users' already-saved collections will silently change contents after upgrading (boundary-day docs disappear).
- `between` exclusive on **both** ends is unusual — in most products "between A and B" includes the boundaries. A user picking Jun 10 – Jun 14 and not seeing Jun 10/14 docs will read this as a new bug, not a fix.

Whether `after`/`before` should exclude the boundary is a product decision, and it's orthogonal to this feature. I'd suggest splitting the boundary-semantics change into a separate issue/PR for maintainers to decide, and keeping the existing semantics here (or at minimum keeping `between` inclusive). Bundling it risks stalling the whole PR.

### 🟡 3. The stale-`now` fix is only half a fix

Moving `now` inside `map()` only helps when the upstream re-emits (i.e., some doc changes). A "last 30 minutes" view left open with no doc activity will never drop docs as they age out. For day granularity this was theoretical; at 30-minute granularity it's a real UX issue. Consider merging a timer source for the relative methods only, e.g.:

```ts
combineLatest([upstream$, timer(0, 30_000)]).pipe(map(([o]) => ...))
```

At minimum, the limitation is worth documenting in the PR.

### 🟡 4. Sub-day options are meaningless for custom date properties

`WorkspacePropertyTypes.date` (user-created date properties) also spreads `DateFilterMethod`, but those store `YYYY-MM-DD` only — the code comment itself notes they resolve to midnight UTC. "Last 30 minutes" on such a property will essentially never match. Consider splitting into a day-granularity method map and a timestamp method map, with only `createdAt`/`updatedAt` getting the sub-day entries.

### 🔵 Minor

- +1 to CodeRabbit's dedup suggestion: `isAfterOrOn` / `isStrictlyAfter` / `isStrictlyBefore` differ only in the final operator — a shared `compareDay(target, ref)` returning -1/0/1 collapses them to one-liners.
- Tests pin the clock to a UTC instant but `dayjs()` uses the local timezone for day/week/month boundaries. Fine on UTC CI, but `this-week`/`this-month` assertions can flip on machines in extreme timezones. Setting `process.env.TZ` in the spec would make it robust.
- The new options are appended after `this-year`, so the dropdown shows "last 30 minutes" at the very bottom. Placing them before `last 3 days` keeps the list ordered short → long.
- `method in relativeMinuteRanges` matches prototype properties (e.g. `toString`); it follows the existing pattern, but `Object.hasOwn` would be safer.

### Not blocking, for later

The original issue (#15368) actually asks for a free-form "newer than X hours Y minutes" input; the six presets cover the common cases but don't fully satisfy that — could be a follow-up reusing the `after`-style value channel. An e2e test exercising the dropdown would also be a nice addition on top of the unit tests.
